crates/node/src/validate.rs

```diff
@@ -27,7 +27,51 @@ struct State {
     block_number: Word,
     solution_index: u64,
     pre_state: bool,
-    conn_pool: db::ConnectionPool,
+    conn_pool: Db,
+}
+
+#[derive(Clone)]
+/// Either a dry run database or a connection pool.
+enum Db {
+    DryRun(DryRun),
+    ConnectionPool(ConnectionPool),
+}
+
+#[derive(Clone)]
+/// A dry run database.
+///
+/// Cascades from in-memory to on-disk database.
+pub struct DryRun {
```

Comment:
Do we need to expose this type publicly? Or is it enough to have some pub `validate_dry_run` fn or similar?